airflow/providers/salesforce/hooks/salesforce.py

```diff
@@ -131,22 +131,39 @@ def get_conn(self) -> api.Salesforce:
         if not self.conn:
             connection = self.get_connection(self.conn_id)
             extras = connection.extra_dejson
+            # all extras below (besides the version one) are explicitly defaulted to None
+            # because simple-salesforce has a built-in authentication-choosing method that
+            # relies on which arguments are None and without "or None" setting this connection
+            # in the UI will result in the blank extras being empty strings instead of None,
+            # which would break the connection if "get" was used on its own.
             self.conn = Salesforce(
                 username=connection.login,
                 password=connection.password,
-                security_token=extras["extra__salesforce__security_token"] or None,
-                domain=extras["extra__salesforce__domain"] or None,
+                security_token=extras.get('security_token')
```

Comment:
question... why are we [re?]introducing the `security_token` alongside the `extra__salesforce__`-formatted one?

is it for backward compat?  i'd assume if these were  renamed then it was done in a major and we need not worry about backward compat?

Reply:
In most cases when defining a connection using a secrets backend, such as via an environment variable, or using a provider's secrets backend, it is not typical to have to put the full "extra__salesforce__security_token" name. Even the connections documentation for the Salesforce provider (https://airflow.apache.org/docs/apache-airflow-providers-salesforce/stable/connections/salesforce.html) shows defining the connection as an environment variable like so:
`export AIRFLOW_CONN_SALESFORCE_DEFAULT='http://your_username:your_password@https%3A%2F%2Fyour_host.lightning.force.com?security_token=your_token'`
which is currently incorrect. Without this PR, you would need to provide it like so (with all of the extras defined:
`export AIRFLOW_CONN_SALESFORCE_DEFAULT='http://your_username:your_password@https%3A%2F%2Fyour_host.lightning.force.com?extra__salesforce__security_token=your_token&extra__salesforce__domain=&extra__salesforce__instance=&extra__salesforce__instance_url=&extra__salesforce__organization_id=&extra__salesforce__version=&extra__salesforce__proxies=&extra__salesforce__client_id=Aiflow&extra__salesforce__consumer_key=&extra__salesforce__private_key_file_path=&extra__salesforce__private_key='`

So, this PR is to allow for `?security_token=your_token` as shown in the documentation, instead of having to do `?extra__salesforce__security_token=your_token`

I can't find another provider that requires the `extra__`-prefixed name to be used when defining the connection via environment variable

Reply:
I see.

I am pretty sure gcp ones use the extra format.  @potiuk or @mik-laj  may be able to help guide us here

Reply:
Maybe I should split bringing in the non-prefixed extras into a seperate PR and not over-complicate this PR?

Reply:
that would probably be best.  i think it would make this PR uncontroversial.

Reply:
also, looking closer at your comment, i'm not opposed to the part of this PR that makes all of the extras optional from secrets backend -- that part we should 100% do.  The part that i think would be inconsistent with past practice is to allow two different keywords for the exact same parameter.

So to use your example, what we I think we should be able to do when this pr is merged is this:
```
AIRFLOW_CONN_SALESFORCE_DEFAULT='http://your_username:your_password@https%3A%2F%2Fyour_host.lightning.force.com?extra__salesforce__security_token=your_token'
```
so you only have to specify the extra you need, _but_ you must use the same keyword as that generated by the UI.

Reply:
looking more closely again:

> In most cases when defining a connection using a secrets backend, such as via an environment variable, or using a provider's secrets backend, it is not typical to have to put the full "extra__salesforce__security_token" name. 

can you provide an example?

i am only aware of GCP and it's not like this.

> Even the connections documentation for the Salesforce provider (https://airflow.apache.org/docs/apache-airflow-providers-salesforce/stable/connections/salesforce.html) shows defining the connection as an environment

You are correct it appears they missed updating the example URI when they added the UI widgets and removed `security_token` as an extra.  The doc is not correct.

Reply:
ok so i did a search of the repo and found 3 other instances where both long name and short name are accepted so i withdraw my objections and let's go with it.

Reply:
I'm not considering this resolved as I would like to move the shorter names to a separate PR. Once I took it out of this PR I will then mark this resolved

Reply:
Ok. I updated this to only include the long extra names. I am going to wait until this gets merged in before making a new PR to adding in the shorter names.

Reply:
Sounds good. Just reviewed